src/librustc/hir/lowering.rs

```diff
@@ -519,7 +518,8 @@ impl<'a> LoweringContext<'a> {
         }
     }
 
-    fn insert_item(&mut self, id: NodeId, item: hir::Item) {
+    fn insert_item(&mut self, item: hir::Item) {
+        let id = item.hir_id;
```

Comment:
Maybe you could try asserting that the local id of the item is 0 here?

You could also try printing out information about the item which isn't 0.

Reply:
Got one:
```
non-zero local id in Item {
	ident: TryFrom#0, 
	hir_id: HirId { owner: DefIndex(0:147), local_id: 1 }, 
	attrs: [], 
	node: Use(path(convert::TryFrom), Single), 
	vis: Spanned { node: Inherited, span: Span { lo: BytePos(73297), hi: BytePos(73297), ctxt: #0 } }, 
	span: Span { lo: BytePos(73311), hi: BytePos(73318), ctxt: #0 }
}
```